applications/matter_weather_station/README.rst

```diff
@@ -143,22 +148,22 @@ Selecting a build type from command line
    :start-after: build_types_selection_cmd_start
    :end-before: For example, you can replace the
```

Comment:
@pdunaj , can we fix this section of gs_modifying.rst in this PR before propagating these build type changes to other applications? These changes improve the build type selection, but they end up creating inconsistencies and issues in the documentation.

Reply:
We can but changes in `gs_modifying.rst` affect everything. I think that the two open PRs are the only one referring to ZDebug/ZRelease style of build type. Once done we can rework the doc. My plan was to fix these two then update doc while checking what can be done for asset tracker (here there is yet another way of defining build type - this time using overlays)